src/contoroller/transaction/transaction.js

```diff
@@ -3,41 +3,36 @@ const _ = require("lodash");
 class TransactionController {
   constructor(Transactions, validator, User, Account, Category) {
     this.Transaction = Transactions;
-    this.validator = validator;
+    this.validateTransaction = validator;
     this.User = User;
     this.Account = Account;
     this.Category = Category;
   }
-  //ok
+
   async create(req, res) {
```

Comment:
Thinks about this scenario:

You receive a request to create a transaction and you find the account and category to attach it to the new transaction. But, in the meantime, you receive a request to delete the category and it is processed faster than the creation of the transaction. 

Then once you continue to create the transaction, the category does not exist anymore and the transaction is not valid anymore. 

These are called race conditions and they can mess up the data in your database, you can use Postgres locks to prevent them.